Find a functionally equivalent code:android.media.AudioManager.registerMediaButtonIntent:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 * no-op if (pi == null) or (eventReceiver == null)
 */

@@ -3,13 +3,8 @@
         Log.e(TAG, "Cannot call registerMediaButtonIntent() with a null parameter");
         return;
     }
-    IAudioService service = getService();
-    try {
-        // pi != null
-        service.registerMediaButtonIntent(pi, eventReceiver, eventReceiver == null ? mToken : null);
-    } catch (RemoteException e) {
-        Log.e(TAG, "Dead object in registerMediaButtonIntent" + e);
-    }
+    MediaSessionLegacyHelper helper = MediaSessionLegacyHelper.getHelper(mContext);
+    helper.addMediaButtonListener(pi, eventReceiver, mContext);
 }

